### ISSUE TITLE
webpack 5 the page opens blank the first time when use lazy compilation

### ISSUE THREAD

**Issue by await-ovo** [webpack/webpack-dev-server#3204]
# Bug report
When I try lazy compilation, I find that the first time I open the page is always blank.
Here is a simple repo to reproduce: [https://github.com/await-ovo/webpack5-lazy](https://github.com/await-ovo/webpack5-lazy)
1. `git clone https://github.com/await-ovo/webpack5-lazy`
2. `cd webpack5-lazy && yarn`
3. `yarn dev`

Then open `localhost:8080/a.html`, you can see that  the page is empty:

![image](https://user-images.githubusercontent.com/41503212/115385828-2a1fd000-a20b-11eb-8332-2f04857ded9b.png)

This is my webpack config:
```
const path = require('path');
const HtmlWebpackPlugin = require('html-webpack-plugin')

module.exports = {
  mode: 'development',
  entry: {
    a: path.resolve('./src/a.js'),
    b: path.resolve('./src/b.js')
  },
  devtool: 'cheap-module-source-map',
  devServer: {
    dev: {
      writeToDisk: true,
    }
  },
  optimization: {
    runtimeChunk: true,
  },
  experiments: {
    lazyCompilation: true,
  },
  plugins: [
    new HtmlWebpackPlugin({
      filename: 'a.html',
      template: path.resolve('./index.html'),
      chunks:['a']
    }),
    new HtmlWebpackPlugin({
      filename: 'b.html',
      chunks:['b'],
      template: path.resolve('./index.html')
    })
  ]
}
```

**What is the current behavior?**
The page opens blank  in the browser and must be refreshed manually

**If the current behavior is a bug, please provide the steps to reproduce.**
Reproduction steps are as described above

**What is the expected behavior?**
It should display content correctly.

**Other relevant information:**
Once the runtimeChunk is commented out, everything works fine.
If it's a bug, I'm happy to help fix it, but I need some guidance, thanks~
webpack version:  5.34
Node.js version: 14
Operating System:  macos
Additional tools:

**Comment by await-ovo** [comment]
> Know issue with dev server

I will take a look at the dev server implementation and would be happy to fix Since I am not familiar with dev server, can you give me some advice, thanks a lot!

**Comment by alexander-akait** [comment]
> There is a race condition in webpack-dev-server which makes it skip updates if the compilation finishes while the WebSocket connection is being established...
> I deterministically run it that, because with lazy compilation the compilation is triggered during page load. If the build is fast enough this problem happens.

Here problem with `initial` https://github.com/webpack/webpack-dev-server/blob/master/client-src/index.js#L23

It should be like `if(status.currentHash.indexOf(__webpack_hash__) < 0) return;` where `__webpack_hash__` is hash of previously build

**Comment by zhangchao828** [comment]
#3232 https://github.com/webpack/webpack/issues/12564

**Comment by alexander-akait** [comment]
Fixed, please update `webpack@5.51.0` and `webpack-dev-server@4.0.0`
